### PR TITLE
Move version check into non-branch condition.

### PR DESCRIPTION
**Pull request by david-mcmahon**
cc @javier-b-perez

### CODE REVIEW

File: anago [hunk 1]
```diff
@@ -756,11 +761,6 @@ get_build_candidate () {
...
   # This is to ensure that branchff can continue to be used as needed.  User can
   # override here.
-  if [[ ! $JENKINS_BUILD_VERSION =~ ${VER_REGEX[release]} ]]; then
-    logecho "Invalid format: $JENKINS_BUILD_VERSION"
-    return 1
-  fi
-
   # For master branch alpha builds, ensure we've released the previous .0 first
   if [[ -z "$PARENT_BRANCH" && -n ${RELEASE_VERSION[alpha]} ]]; then
     latest_official=${BASH_REMATCH[1]}.$((${BASH_REMATCH[2]}-1))
```
Comment (ixdy) [inline anago]: this depends on the bit you moved, right? are there cases this would not be set, now?

Comment (david-mcmahon) [inline anago]: Yeah, really.  Keeping that whole thing as a unit now in branchff_sanity_check().

**Comment by david-mcmahon** [comment]
PTAL

**Comment by david-mcmahon** [comment]
ping

**Comment by enisoc** [comment]
What's the overall reason for moving this check?

**Comment by david-mcmahon** [comment]
The original check was outside of the scope/condition it belonged in.  @ixdy pointed out that my initial change ignored some other related code, so I just pulled it all into a function so it was easier to see all the logic as a unit.

**Pull request merged**